### PR TITLE
baitap14

### PR DESCRIPTION
**Pull request by Dinhduy01**
## Related Tickets
- [#Task xxxxx: Content](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
- Bài tập Shared Preferences
## Evidence (Screenshot or Video)
096c45e5e96c3132687d
8a2183b72f3ef760ae2f

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/android/bai14_share/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    override fun onClick(p0: View?) {
+        when (p0) {
+            binding.buttonBlack -> changebackground(p0)
+            binding.buttonRed -> changebackground(p0)
+            binding.buttonGreen -> changebackground(p0)
+            binding.buttonBlue -> changebackground(p0)
+            binding.buttonReset -> reset()
+            binding.countButton -> countUp()
+        }
```
Comment (hungld-1302) [inline app/src/main/java/com/sun/android/bai14_share/MainActivity.kt]:
```suggestion
        when (p0) {
            binding.buttonBlack,binding.buttonRed,binding.buttonGreen,binding.buttonBlue -> changebackground(p0)
            binding.buttonReset -> reset()
            binding.countButton -> countUp()
        }
```

File: app/src/main/java/com/sun/android/bai14_share/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        mCount = mPreferences.getInt(COUNT_KEY, 0)
+        mShowCountTextView.text = String.format("%s", mCount)
+
+
+
+        mColor = mPreferences.getInt(COLOR_KEY, mColor)
+        mShowCountTextView.setBackgroundColor(mColor)
+
+
+
```
Comment (hungld-1302) [inline app/src/main/java/com/sun/android/bai14_share/MainActivity.kt]: cachs1 dòng là được r em

File: app/src/main/res/layout/activity_main14.xml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    <androidx.constraintlayout.widget.Guideline
+        android:id="@+id/guideline_lower"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        app:layout_constraintGuide_end="94dp" />
+
+    <Button
+        android:id="@+id/count_button"
```
Comment (hungld-1302) [inline app/src/main/res/layout/activity_main14.xml]: .

File: app/src/main/java/com/sun/android/bai14_share/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.sun.android.R
+import com.sun.android.databinding.ActivityMain14Binding
+
+
+class MainActivity : AppCompatActivity(), View.OnClickListener {
+    val binding by lazy { ActivityMain14Binding.inflate(layoutInflater) }
+    lateinit var mPreferences: SharedPreferences
```
Comment (ghost) [inline app/src/main/java/com/sun/android/bai14_share/MainActivity.kt]: dùng lazy

File: app/src/main/java/com/sun/android/bai14_share/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    fun reset() {
+        mCount = 0
+        mShowCountTextView.text = String.format("%s", mCount)
+        mColor = ContextCompat.getColor(
+            this, R.color.default_background
+        )
+        mShowCountTextView.setBackgroundColor(mColor)
+        val preferencesEditor = mPreferences!!.edit()
```
Comment (ghost) [inline app/src/main/java/com/sun/android/bai14_share/MainActivity.kt]: không dùng assert non null

File: app/src/main/java/com/sun/android/bai14_share/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    fun changebackground(view: View) {
+
+        view.backgroundTintList?.let {
+            mShowCountTextView.setBackgroundColor(
+                it.getColorForState(
+                    intArrayOf(android.R.attr.state_enabled), 0
+                )
+            )
+        }
+        mColor = ((mShowCountTextView.background) as ColorDrawable).color
```
Comment (ghost) [inline app/src/main/java/com/sun/android/bai14_share/MainActivity.kt]:
```suggestion
        }
        mColor = ((mShowCountTextView.background) as? ColorDrawable)?.color
```

**Pull request merged**